### ISSUE TITLE
`internal error: parent group` when using `placement: auto` in `figure` for some large content

### ISSUE THREAD

**Issue by nik-nul** [typst/typst#8487]
### Description

```
error: internal error: parent group (occurred at crates/typst-pdf/src/tags/tree/build.rs:202:14)
 = hint: please report this as a bug
```
<img width="827" height="520" alt="Image" src="https://github.com/user-attachments/assets/4866840c-e53a-4c53-a57b-75f61ff39563" />

Compiling this document under typst 0.14.2 also give the same error message.

If I remove one `figure(..., placement: auto)` it will give no error message.

Here is the "minimum" reproduction code I could ever found, and every single line is necessary:
```typ
#import "@preview/cetz:0.3.4"
#set text(font: ("Libertinus Serif", "Noto Serif CJK SC")) // Noto Serif CJK SC is necessary
#let d = json("./ice.json")
#let f(m) = {
  let a = (m.at(0),)
  for c in m.slice(1) {
    a.push(f(c))
  }
  a
}
#{
  heading(level: 1, "")
  v(140.25pt)
  for s in (
    ("", d.p.e),
    ("", d.p.a),
    ("", d.p.b),
    ("", d.x),
    ("", d.p.c),
    ("", d.p.d),
  ) {
    let (h, c) = s
    heading(level: 2, h)
    c
  }
  heading(level: 2, "")
  figure(cetz.canvas({
    cetz.tree.tree(f(d.t), direction: "right", spread: 1.5, grow: 3.5)
  }), placement: auto)
  heading(level: 2, "")
  for i in d.g {
    heading(level: 2, "")
    [\ *A*]
    let t2 = ()
    for c in i.c {
      t2.push(c.a)
      t2.push(c.b)
      t2.push(c.c)
      t2.push(c.d)
      t2.push(c.e)
    }
    table(
      columns: (auto, auto, auto, auto, auto),
      table.header([一一一一]),
      ..t2,
    )
  }
}
```
I will upload `ice.json` later.

### Operating system

Linux

### Typst version

- [x] I am using the latest version of Typst

**Comment by nik-nul** [comment]
Here is the link of ice.json: 
[https://box.nju.edu.cn/f/676a925c668843338720/](https://box.nju.edu.cn/f/676a925c668843338720/)

Since it is a large file, I don't know the best way to upload it. The link above will be available for at least one year.

<img width="827" height="520" alt="Image" src="https://github.com/user-attachments/assets/e086aa76-ab01-4950-8971-28fa4a6d034f" />

**Comment by saecki** [comment]
I've reduced this to the following MRE:

```typ
--- issue-8487-missing-parent-broken-table-with-figure-auto-placment paged ---
#set page(width: 120pt, height: 95pt, margin: 10pt)
#rect(height: 20pt)
#figure(placement: auto, rect(height: 50pt))
#table(
  columns: 2,
  [A], lorem(2),
  [B], lorem(10),
  [C], lorem(2),
)
```
Which produces the following output:
<img width="160" height="383" alt="Image" src="https://github.com/user-attachments/assets/49fb7b16-f02c-4942-ae4d-8b55151cc1a1" />

The `"B"` and `"Lorem ipsum"` of the second cell is completely missing, which seems like a bug in the grid layouter.

**Comment by deivisCode** [comment]
I got a similar `internal error: parent group (occurred at crates/typst-pdf/src/tags/tree/build.rs:202:14)` while migrating our journal to typst.

I'm not using any table or figure (like the MRE from Saecki) so hopefully sharing the code here is helpfull. Unfortunately, minimizing the code made the crash so brittle that literally changing a character somewhere fixes it. I'm afraid I cant reduce it significantly more, at this point it was just deleting random lines or swapping some paragraphs to `lorem's` for 3h to no avail. If there is any tip for cleaning it, Im open to it.

It also needs an external bibliography, fonts and a custom CSL style (I couldn't get rid of them, sorry) so I just made an orphan branch as a stripped down version of our journal with only one `*.typ` and the necessary files.

Im compiling with latest Typst in Ubuntu and Windows 11.

### CODE
https://github.com/deivisCode/revista/tree/crash

**Comment by saecki** [comment]
> I got a similar `internal error: parent group (occurred at crates/typst-pdf/src/tags/tree/build.rs:202:14)` while migrating our journal to typst.
> 
> I'm not using any table or figure (like the MRE from Saecki) so hopefully sharing the code here is helpfull. Unfortunately, minimizing the code made the crash so brittle that literally changing a character somewhere fixes it. I'm afraid I cant reduce it significantly more, at this point it was just deleting random lines or swapping some paragraphs to `lorem's` for 3h to no avail. If there is any tip for cleaning it, Im open to it.
> 
> It also needs an external bibliography, fonts and a custom CSL style (I couldn't get rid of them, sorry) so I just made an orphan branch as a stripped down version of our journal with only one `*.typ` and the necessary files.
> 
> Im compiling with latest Typst in Ubuntu and Windows 11.
> ### CODE
> 
> https://github.com/deivisCode/revista/tree/crash

This actually does seem to be the same problem. The bibliography uses a grid internally and here the last cell seems to be broken across pages, the second image shows part of the URL of the `cygnus_x1` citation. But the first part including the cell is missing.

<table>
<tr>
<td width="50%">
<img width="1191" height="1684" alt="Image" src="https://github.com/user-attachments/assets/fa532151-9b69-4811-a9a7-59b17111f741" />
</td>
<td width="50%">
<img width="1191" height="1684" alt="Image" src="https://github.com/user-attachments/assets/ca3bec77-fbbb-4c4e-af90-070345046269" />
</td>
</tr>
</table>
